bind-contrib/contrib/updatehosts-1.1.0/man/cvtstatic.1
.\" $Id: cvtstatic.1,v 1.1 2000/07/11 06:53:18 vixie Exp $
.TH CVTSTATIC 1 $Date: 2000/07/11 06:53:18 $
.UC 4
.SH NAME
cvtstatic \- convert an /etc/hosts file to updatehosts database format
.SH SYNOPSIS
.B cvtstatic
[
.B -a
]
[
-c
.B -d default_domain
[
.B -h
]
[
.B -i ip_address_prefix
]
.B -o output_directory
.SH DESCRIPTION
.PP
.B cvtstatic
is a utility to convert an existing /etc/hosts file to
.B updatehosts
format.
The hosts file is read from stdin.
The output is a set of
.B updatehosts
relations located in the directory specified by the -d option.
.PP
The -a will append output to an already existing set of relations.
.PP
The -c option will add comments from the zone file.
The comments are placed in the same location they are found in the zone
file.
.PP
The -d option is required and specifies the default domain name for this
set of input records.
.PP
The -h option outputs a short command synopsis.
.PP
The -i option specifies an option IP address prefix for the IP addresses
found in A records.
.PP
The -o option which is required specifies a directory where the
.B updatehosts relations are found.
.SH
Readinfo Record Generation
.PP
.B cvtstatic
generates
.B readinfo
records using the default domain and the IP address prefix to
generate a correct #FIELDS line which allows domain names and IP addresses
to be written in a shorthand manner.
All domains are shortened by chopping off the default domain name suffix.
Those domain names which are in a different domain are written fully
qualified with a trailing period.
.PP
If the -i option is present all IP addresses which start with the IP prefix
have it removed.
All other IP address will be written as a standard dotted decimal IP
address with a leading period (".").
The leading period denotes that the default IP prefix is not prepended to
the address.
Without the -i option all IP addresses are written in standard
dotted-decimal notation and there is no default prefix for IP addresses.
.PP
Each invocation of
.B cvtstatic
generates a new
.B readinfo
#FIELDS line.
This is convenient when converting multiple hosts files where the default
domain or the default IP prefix have different values.
.SH
See also
.PP
.B updatehosts(1), gendns(1), readinfo(1)


CHANGES
5500.	[bug]		Fake entry.

5494.	[bug]		Silence the EPROTO syslog message on older systems.
			[GL #1928]

5493.	[bug]		Fix off-by-one error when calculating new hashtable
			size. [GL #2104]

5492.	[bug]		Tighten LOC parsing to reject period and/or m as a
			value.  Correct handling of negative altitudes which
			are not whole metres.  [GL #2074]

5491.	[bug]		rbtversion->glue_table_size could be read without the
			appropriate lock being held. [GL #2080]

5490.	[func]		Refactor the readline support to use pkg-config and
			add support for editline library. [GL !3942]

5489.	[bug]		Named failed to reject some invalid records resulting
			in records that, after being printed, could not be
			loaded or would result in DNSSEC validation failures
			when re-read from zone files as the wire format
			differed.  The covered records records are: CERT,
			IPSECKEY, NSEC3, NSEC3PARAM, NXT, SIG, TLSA, WKS, and
			X25. [GL !3953]

5488.	[bug]		nta needed to have a weak reference on view to prevent
			the view being deleted while nta tests are being
			performed. [GL #2067]

5487.	[cleanup]	Update managed keys log messages to be less confusing.
			[GL #2027]

5486.	[func]		Add 'rndc dnssec -checkds' command to tell named
			that the DS record has been published in the parent.
			[GL #1613]

	--- 9.17.4 released ---

5485.	[placeholder]

5484.	[func]		Expire zero TTL records quickly rather than using them
			for stale answers. [GL #1829]

5483.	[func]		Keeping "stale" answers in cache has been disabled by
			default and can be re-enabled with a new configuration
			option "stale-cache-enable". [GL #1712]

5482.	[bug]		If the Duplicate Address Detection (DAD) mechanism had
			not yet finished after adding a new IPv6 address to the
			system, BIND 9 would fail to bind to IPv6 addresses in a
			tentative state. [GL #2038]

5481.	[security]	"update-policy" rules of type "subdomain" were
			incorrectly treated as "zonesub" rules, which allowed
			keys used in "subdomain" rules to update names outside
			of the specified subdomains. The problem was fixed by
			making sure "subdomain" rules are again processed as
			described in the ARM. (CVE-2020-8624) [GL #2055]

5480.	[security]	When BIND 9 was compiled with native PKCS#11 support, it
			was possible to trigger an assertion failure in code
			determining the number of bits in the PKCS#11 RSA public
			key with a specially crafted packet. (CVE-2020-8623)
			[GL #2037]

5479.	[security]	named could crash in certain query resolution scenarios
			where QNAME minimization and forwarding were both
			enabled. (CVE-2020-8621) [GL #1997]

5478.	[security]	It was possible to trigger an assertion failure by
			sending a specially crafted large TCP DNS message.
			(CVE-2020-8620) [GL #1996]

5477.	[bug]		The idle timeout for connected TCP sockets, which was
			previously set to a high fixed value, is now derived
			from the client query processing timeout configured for
			a resolver. [GL #2024]

5476.	[security]	It was possible to trigger an assertion failure when
			verifying the response to a TSIG-signed request.
			(CVE-2020-8622) [GL #2028]

5475.	[bug]		Wildcard RPZ passthru rules could incorrectly be
			overridden by other rules that were loaded from RPZ
			zones which appeared later in the "response-policy"
			statement. This has been fixed. [GL #1619]

5474.	[bug]		dns_rdata_hip_next() failed to return ISC_R_NOMORE
			when it should have. [GL !3880]

5473.	[func]		The RBT hash table implementation has been changed
			to use a faster hash function (HalfSipHash2-4) and
			Fibonacci hashing for better distribution. Setting
			"max-cache-size" now preallocates a fixed-size hash
			table so that rehashing does not cause resolution
			brownouts while the hash table is grown. [GL #1775]

5472.	[func]		The statistics channel has been updated to use the
			new network manager. [GL #2022]

5471.	[bug]		The introduction of KASP support inadvertently caused
			the second field of "sig-validity-interval" to always be
			calculated in hours, even in cases when it should have
			been calculated in days. This has been fixed. (Thanks to
			Tony Finch.) [GL !3735]

5470.	[port]		gsskrb5_register_acceptor_identity() is now only called
			if gssapi_krb5.h is present. [GL #1995]

5469.	[port]		On illumos, a constant called SEC is already defined in
			<sys/time.h>, which conflicts with an identically named
			constant in libbind9. This conflict has been resolved.
			[GL #1993]

5468.	[bug]		Addressed potential double unlock in process_fd().
			[GL #2005]

5467.	[func]		The control channel and the rndc utility have been
			updated to use the new network manager. To support
			this, the network manager was updated to enable
			the initiation of client TCP connections. Its
			internal reference counting has been refactored.

			Note: As a side effect of this change, rndc cannot
			currently be used with UNIX-domain sockets, and its
			default timeout has changed from 60 seconds to 30.
			These will be addressed in a future release.
			[GL #1759]

5466.	[bug]		Addressed an error in recursive clients stats reporting.
			[GL #1719]

5465.	[func]		Added fallback to built-in trust-anchors, managed-keys,
			or trusted-keys if the bindkeys-file (bind.keys) cannot
			be parsed. [GL #1235]

5464.	[bug]		Requesting more than 128 files to be saved when rolling
			dnstap log files caused a buffer overflow. This has been
			fixed. [GL #1989]

5463.	[placeholder]

5462.	[bug]		Move LMDB locking from LMDB itself to named. [GL #1976]

5461.	[bug]		The STALE rdataset header attribute was updated while
			the write lock was not being held, leading to incorrect
			statistics. The header attributes are now converted to
			use atomic operations. [GL #1475]

5460.	[cleanup]	tsig-keygen was previously an alias for
			ddns-confgen and was documented in the ddns-confgen
			man page. This has been reversed; tsig-keygen is
			now the primary name. [GL #1998]


bind-contrib/contrib/updatehosts-1.1.0/cvtstatic.pl
#! @PERL@
#
# $Id: cvtstatic.pl,v 1.1 2000/07/11 06:52:51 vixie Exp $

# updatehosts DNS maintenance package
# Copyright (C) 1998-1999  Smoot Carl-Mitchell
# 
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
# 
# smoot@tic.com

# convert static host table to updathost file format
#

require "getopts.pl";

Getopts("acd:hi:o:");
if ($opt_h || ! $opt_d || ! $opt_o) {
	print STDERR <<EOF;
usage cvtstatic [-a] [-c] -d domain [-i IPprefix] -o directory
	-a - append to output files instead of overwriting
	-c - include comments from zone file
	-d - default domain for output
	-h - this help message
	-i - IP address prefix
	-o - directory to write output
EOF
	exit;
}
$append = $opt_a;
$app = ">";
$app = ">>" if $append;
$comments = $opt_c;
$dbdir = $opt_o;
$default_domain = $opt_d;
$default_domain .= "." if $default_domain !~ /\.$/;
$domain_suffix = $default_domain; chop($domain_suffix);	# Remove trailing dot.
$domain_suffix = ".$domain_suffix" if $domain_suffix !~ /^\./;
$ip_prefix = $opt_i;
$ip_prefix .= "." if $ip_prefix && $ip_prefix !~ /\.$/;

# open up the database files
open(MAIN, "$app$dbdir/main") || die "cannot open $dbdir/main";
open(CNAME, "$app$dbdir/cname") || die "cannot open $dbdir/cname";

if ($domain_suffix || $ip_prefix) {
	print MAIN "#FIELDS GLOBAL null=X host";
	print MAIN " suffix=$domain_suffix" if $domain_suffix;
	print MAIN " no=. ip";
	print MAIN " no=. prefix=$ip_prefix" if $ip_prefix;
	print MAIN " ether hard os contact norev\n";
	print CNAME "#FIELDS GLOBAL null=X host no=.";
	print CNAME " suffix=$domain_suffix" if $domain_suffix;
	print CNAME " alias no=.";
	print CNAME " suffix=$domain_suffix" if $domain_suffix;
	print CNAME "\n";
}
$ippat = $ip_prefix;
$domainpat = $domain_suffix;
$ippat =~ s/(\W)/\\$1/g;
$domainpat =~ s/(\W)/\\$1/g;
while (<>) {
	chop;

	next if /^\s*$/;

	# single line comments
	if (/^#/) {
		print MAIN "$_\n";
		next;
	}
	# print trailing comments before they are stripped
	if (/#/) {
		s/#.*$//;
		$comment = $1;
		print MAIN "$comment\n";
	}
		
	($ip, $canonical, @cname) = split;
	
	if ($ippat && $ip =~ /^$ippat/) {
		$ip =~ s/^$ippat//;
	}
	else {
		$ip = ".$ip" if $ip_prefix;
	}
	if ($canonical =~ /\./) {
		if ($domainpat && $canonical =~ /$domainpat$/) {
			$canonical =~ s/$domainpat$//;
		}
		else {
			$canonical = "$canonical.";
		}
	}
	print MAIN "$canonical\t$ip\n";
	foreach $cname (@cname) {
		if ($canonical =~ /\./) {
			if ($domainpat && $cname =~ /$domainpat$/) {
				$cname =~ s/$domainpat$//;
			}
			else {
				$cname = "$cname.";
			}
		}
		print CNAME "$canonical\t$cname\n";
	}
}
